# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/components/sessionstore/test/unit/test_migration_lz4compression.js
# Commit: 2a8633af0279
# Full Hash: 2a8633af0279379685ba781c6c469e8b07ec13f4
# Author: Mathew Hodson <mathew.hodson@gmail.com>
# Date: 2022-03-26 09:28:42
# Regressor Bug: 1752853
# File Overlap Count: 1
# Description:
#   Bug 1752853 - Stop using a worker to write session store. r=Gijs,Standard8
#   
#   Remove now unused pref "browser.sessionstore.max_write_failures".
#   Rename and simplify test_shutdown_cleanup.js to make it more clear that
#   it is testing the isFinalWrite option.
# ==============================================================================

diff -r dd81d58d9d5f -r 2a8633af0279 browser/components/sessionstore/test/unit/test_migration_lz4compression.js
--- a/browser/components/sessionstore/test/unit/test_migration_lz4compression.js	Fri Mar 25 19:19:15 2022 +0000
+++ b/browser/components/sessionstore/test/unit/test_migration_lz4compression.js	Fri Mar 25 19:42:36 2022 +0000
@@ -1,14 +1,15 @@
 "use strict";
 
-const { XPCOMUtils } = ChromeUtils.import(
-  "resource://gre/modules/XPCOMUtils.jsm"
-);
-const { SessionWorker } = ChromeUtils.import(
-  "resource:///modules/sessionstore/SessionWorker.jsm"
+const { SessionWriter } = ChromeUtils.import(
+  "resource:///modules/sessionstore/SessionWriter.jsm"
 );
 
-var Paths;
-var SessionFile;
+// Make sure that we have a profile before initializing SessionFile.
+const profd = do_get_profile();
+const { SessionFile } = ChromeUtils.import(
+  "resource:///modules/sessionstore/SessionFile.jsm"
+);
+const Paths = SessionFile.Paths;
 
 // We need a XULAppInfo to initialize SessionFile
 const { updateAppInfo } = ChromeUtils.import(
@@ -37,39 +38,24 @@
 function promise_check_contents(path, expect) {
   return (async function() {
     info("Checking whether " + path + " has the right contents");
-    let actual = await IOUtils.readUTF8(path, {
+    let actual = await IOUtils.readJSON(path, {
       decompress: true,
     });
     Assert.deepEqual(
-      JSON.parse(actual),
+      actual,
       expect,
       `File ${path} contains the expected data.`
     );
   })();
 }
 
-function generateFileContents(id) {
-  let url = `http://example.com/test_backup_once#${id}_${Math.random()}`;
-  return { windows: [{ tabs: [{ entries: [{ url }], index: 1 }] }] };
-}
-
 // Check whether the migration from .js to .jslz4 is correct.
 add_task(async function test_migration() {
-  // Make sure that we have a profile before initializing SessionFile.
-  let profd = do_get_profile();
-  SessionFile = ChromeUtils.import(
-    "resource:///modules/sessionstore/SessionFile.jsm"
-  ).SessionFile;
-  Paths = SessionFile.Paths;
-
   let source = do_get_file("data/sessionstore_valid.js");
   source.copyTo(profd, "sessionstore.js");
 
   // Read the content of the session store file.
-  let sessionStoreUncompressed = await IOUtils.readUTF8(
-    Paths.clean.replace("jsonlz4", "js")
-  );
-  let parsed = JSON.parse(sessionStoreUncompressed);
+  let parsed = await IOUtils.readJSON(Paths.clean.replace("jsonlz4", "js"));
 
   // Read the session file with .js extension.
   let result = await SessionFile.read();
@@ -99,17 +85,16 @@
 
 add_task(async function test_startup_with_compressed_clean() {
   let state = { windows: [] };
-  let stateString = JSON.stringify(state);
 
   // Mare sure we have an empty profile dir.
   await SessionFile.wipe();
 
   // Populate session files to profile dir.
-  await IOUtils.writeUTF8(Paths.clean, stateString, {
+  await IOUtils.writeJSON(Paths.clean, state, {
     compress: true,
   });
   await IOUtils.makeDirectory(Paths.backups);
-  await IOUtils.writeUTF8(Paths.cleanBackup, stateString, {
+  await IOUtils.writeJSON(Paths.cleanBackup, state, {
     compress: true,
   });
 
@@ -155,7 +140,7 @@
 
   // Create a state to store.
   let state = { windows: [] };
-  await SessionWorker.post("write", [state, { isFinalWrite: true }]);
+  await SessionWriter.write(state, { isFinalWrite: true });
 
   // Check session files are created, but not deprecated ones.
   await promise_check_exist(Paths.clean, true);